lib/eslint.js

```diff
@@ -505,6 +506,53 @@ function stripUnicodeBOM(text) {
     return text;
 }
 
+/**
+ * Wraps a rule action with an exception handler.
+ *
+ * The exception handler will catch exceptions thrown by rule actions, decorate
+ * the exception with rule metadata, and rethrow the exception. This will allow
+ * us to detect the uncaught exception as a rule crash and show a user-friendly
+ * error message.
+ *
+ * @param {function} action The rule action to wrap.
+ * @param {object} metadata Metadata that are useful for bug reports.
+ * @returns {function} A function which will catch, decorate, and rethrow rule exceptions.
+ */
+function createRuleExceptionHandler(action, metadata) {
+    var ruleName = metadata.ruleName,
+        fileName = metadata.fileName,
+        parserPath = metadata.parserPath,
+        hasDefaultParser = parserPath === DEFAULT_PARSER,
+        pluginDelimiterIndex = ruleName.indexOf("/"),
+        pluginName = null;
+
+    if (pluginDelimiterIndex > -1) {
+        pluginName = ruleName.slice(0, ruleName.indexOf("/"));
+    }
+
+    return function() {
+        try {
```

Comment:
I don't know of any other way of doing this (other then global error handler), but this will most likely have a major impact on performance. Could you run `npm run perf` before and after this change?